### PR TITLE
[MECHS] Add mecha melee weapons (hammer and sword) & overlay for when the components are damaged

### PR DESCRIPTION
**Pull request by Comicao1**
## About The Pull Request
Add mecha melee weapons
- Hammer (blunt)
- Sword (slash)
- Overlay for when the components are damaged
## Why It's Good For The Game
More options for combat, since player's only available melee weapon is a drill, (and it doesn't really makes sense), so it could be both for antagonists and people combating the antagonists.
Available for: Paddy, Gygax, Durand, Phazon and the HONK mech
## Changelog
:cl:
add: Added new hammer weapon for mechs
add: Added new sword weapon for mechs
add: New effects for when the tools in the mech's hand's are damaged.
/:cl:

### CODE REVIEW

**Comment by DaCoolBoss** [comment]
That hammer looks pretty close to a pickaxe, maybe it should get digging ability?

File: code/modules/vehicles/mecha/equipment/weapons/sword.dm [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	do_attack_animation(target, ATTACK_EFFECT_SLASH)
+
+	var/splatter_dir = get_dir(chassis, target)
+	if(isalien(target))
+		new /obj/effect/temp_visual/dir_setting/bloodsplatter/xenosplatter(target.drop_location(), splatter_dir)
+	else
+		new /obj/effect/temp_visual/dir_setting/bloodsplatter(target.drop_location(), splatter_dir)
+
+	if(target_part && prob(10))
+		target_part.dismember(BRUTE)
```
Comment (SmArtKar) [inline code/modules/vehicles/mecha/equipment/weapons/sword.dm:60]: This directly applies damage, that's something you should avoid. This won't trigger any defenses like reactive armors, and dismember will happen regardless of whenever the hit was blocked or not.

File: code/modules/vehicles/mecha/equipment/weapons/hammer.dm [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	target.throw_at(throw_target, 4, 4)
+
+	var/splatter_dir = get_dir(chassis, target)
+	if(isalien(target))
+		new /obj/effect/temp_visual/dir_setting/bloodsplatter/xenosplatter(target.drop_location(), splatter_dir)
+	else
+		new /obj/effect/temp_visual/dir_setting/bloodsplatter(target.drop_location(), splatter_dir)
+
+	if(target_part && prob(10))
+		target_part.dismember(BRUTE)
```
Comment (SmArtKar) [inline code/modules/vehicles/mecha/equipment/weapons/hammer.dm:64]: Same here

**Comment by Jacquerel** [comment]
Your PR will likely meet with more success if you can outline why it's good to have melee weapons attached to mechs, with more specificity than you have done right now.
Where are they useful? What situations do you expect them to be used in? How common do you expect them to be?

**Comment by Potato-Masher** [comment]
I would like to note that some of the combat mechs already have powerful melee attacks, the Gygax and Durand in particular come to mind. Dedicated arm slot melee weapons would have to be either pretty good compared to a raw punch or allow some new tactic that just mecha punching people doesn't provide normally to make sense choosing over a mech LMG or similar.

**Comment by GuillaumePrata** [comment]
> Dedicated arm slot melee weapons would have to be either pretty good compared to a raw punch or allow some new tactic that just mecha punching people doesn't provide normally

We simply nerf mech punching and move those functionalities to mech melee weapons.

Stuff like breaking down walls/airlocks could be a hammer use as an example.

**Comment by carlarctg** [comment]
melee weaponson mechs is a PERFECT opportunity for area of effect atacks, like a 1x3 sweep with a broadsword (horizontal) or longsword (vertical) if this pr doesnt make use of that ill be very disappointed

**Comment by githubuser4141** [comment]
> Your PR will likely meet with more success if you can outline why it's good to have melee weapons attached to mechs, with more specificity than you have done right now. Where are they useful? What situations do you expect them to be used in? What mechs can they be attached to? How common do you expect them to be?
> 
> Right now it's not clear exactly what the use case of these are (except I guess "killing the crew as a traitor") or why the crew would want a short ranged weapon instead of a gun.

Cuz it's a dedicated melee weapon that takes away using a gun. It forces the driver to chose between a powerful melee ability or being able to attack things at range. I'd guess availability is done via RND, the same way mech guns are currently obtained.

**Comment by GPeckman** [comment]
Running the numbers, these weapons seem to be a straight downgrade compared to the built-in melee attacks that most mechs have. At 25 damage each, only the Phazon (15 damage) and Paddy (no melee attack IIRC) have lower damage than the new weapons. The Gygax has equal damage, and the Honker and Durand have higher damage at 30 and 40 respectively.

Furthermore, the cooldown for the new weapons is noticeably higher than the built-in melee cooldown, 1.5 seconds vs 1 second. Given that these weapons will be competing with actual guns for a spot on a mech, and suggest giving them a sizable buff. More damage/faster attack speed would be a nice start, but having a little bit of AoE attack like carlarc suggested is also a very good idea.

**Comment by zoomachina** [comment]
Does this really suit a paddy mech?

**Comment by CCC23** [comment]
finally I can play lancer in ss13

**Comment by SimplySmiley123** [comment]
> > Your PR will likely meet with more success if you can outline why it's good to have melee weapons attached to mechs, with more specificity than you have done right now. Where are they useful? What situations do you expect them to be used in? What mechs can they be attached to? How common do you expect them to be?
> > Right now it's not clear exactly what the use case of these are (except I guess "killing the crew as a traitor") or why the crew would want a short ranged weapon instead of a gun.
> 
> Cuz it's a dedicated melee weapon that takes away using a gun. It forces the driver to chose between a powerful melee ability or being able to attack things at range. I'd guess availability is done via RND, the same way mech guns are currently obtained. IMHO, this should completely replace bare-handed mech melee, which currently doesn't make sense anyway.
> 
> It also allows proper values to be set for mech melee (demolition mod, armor pen and damage), plus allows for special attack reactions (afterattacks in code?) rather then just a 'force variable'.
> 
> Oh, and it means melee is a an actual... weapon, one that's examineable and can be damaged by targeting the arms.

The mech melee does make sense, it's still a hulking piece of machinery with arms (for holding weapons, but still)
Getting smacked across the face by a giant combat mech should still hurt

**Comment by githubuser4141** [comment]
> > > Your PR will likely meet with more success if you can outline why it's good to have melee weapons attached to mechs, with more specificity than you have done right now. Where are they useful? What situations do you expect them to be used in? What mechs can they be attached to? How common do you expect them to be?
> > > Right now it's not clear exactly what the use case of these are (except I guess "killing the crew as a traitor") or why the crew would want a short ranged weapon instead of a gun.
> > 
> > 
> > Cuz it's a dedicated melee weapon that takes away using a gun. It forces the driver to chose between a powerful melee ability or being able to attack things at range. I'd guess availability is done via RND, the same way mech guns are currently obtained. IMHO, this should completely replace bare-handed mech melee, which currently doesn't make sense anyway.
> > It also allows proper values to be set for mech melee (demolition mod, armor pen and damage), plus allows for special attack reactions (afterattacks in code?) rather then just a 'force variable'.
> > Oh, and it means melee is a an actual... weapon, one that's examineable and can be damaged by targeting the arms.
> 
> The mech melee does make sense, it's still a hulking piece of machinery with arms (for holding weapons, but still) Getting smacked across the face by a giant combat mech should still hurt

That's assuming there's something to smack you with, it could be a bare socket that doesn't move or has a very weak actuator that can't move with enough force to strike anything. I like to imagine it's just a socket with some connections capable of weakly moving about to adjust itself

**Comment by mc-oofert** [comment]
these are less useful than actually just punching someone with a real combat mecha
like why would anyone even use these when you could have a gun in that same slot and still be able to beat people to death really easily because combat mechas have really good punches

**Comment by moocowswag** [comment]
There are like a dozen ways you could go about making a melee mech weapon desirable for both crew and traitors without making them oppressive, pick 2 of the following

1. Extremely low power usage so it can be used in very very drawn out combat.
2. Can block a low-medium % of hits that would otherwise hit the mech if facing enemy
3. Can swing and hit enemies in an arc next to the mech unless they are resting (dark souls boss mech that people have to wave dash to beat in melee)
4. Counters heavy armor so it can beat other mechs


There is a rather large power budget though because you are putting a melee weapon on the slowest and most telegraphed thing in the game.

**Comment by Loafin34** [comment]
give hammer the durand KO or let us spin2win it

**Comment by Exester509** [comment]
> > Dedicated arm slot melee weapons would have to be either pretty good compared to a raw punch or allow some new tactic that just mecha punching people doesn't provide normally
> 
> We simply nerf mech punching and move those functionalities to mech melee weapons.
> 
> Stuff like breaking down walls/airlocks could be a hammer use as an example.

Yeah, let's make the least versatile tool in the game even less versatile. That sounds like something that will make mechs more enjoyable for sure.

**Comment by kittysmooch** [comment]
lots of good suggestions already but you should also give the sword sharpness and a butchering component because it'd be funny :3

**Comment by GuillaumePrata** [comment]
> Yeah, let's make the least versatile tool in the game even less versatile. That sounds like something that will make mechs more enjoyable for sure.

There is one concept called "give and take"
Remove the boring punches, give them cool melee weapons which broaden their possibilities.

**Comment by MoonDeer728** [comment]
why not for ripley?

**Comment by Exester509** [comment]
> > Yeah, let's make the least versatile tool in the game even less versatile. That sounds like something that will make mechs more enjoyable for sure.
> 
> There is one concept called "give and take" Remove the boring punches, give them cool melee weapons which broaden their possibilities.

Those "boring punches" let mechs be actually useful and punishing in both melee and ranged combat. A single gun mech is even more of a joke than dual gun ones, and those are bad enough already

**Comment by KingkumaArt** [comment]
> Give Gygax a boost to punches when boosting, either damage or action speed, so Gygax maintains it's dynamic as Mr Lister the Mecha Fister. 

the concept of a no-gun gygax, a mech with already kinda mediocre melee armor, being a melee-only glass cannon sounds kickass.

**Comment by Loafin34** [comment]
make paddy fists do stamina damage (non-lethal guys)

**Comment by projectkepler-ru** [comment]
> lots of good suggestions already but you should also give the sword sharpness and a butchering component because it'd be funny :3

abysmal take

**Comment by projectkepler-ru** [comment]
> make paddy fists do stamina damage (non-lethal guys)

equally aby- wait hold on I'll do that

**Comment by Jacquerel** [comment]
Make a new PR if you ever get around to doing any more work on this

**Pull request closed without merging**